dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordWriter.java

```diff
@@ -415,10 +415,34 @@ else if (fd.getDataLen() < 8 && fd.getDataLen() > 0)
                 {
                     value = (Double) fieldValue;
                 }
-                else if (fieldValue instanceof Integer)
+                else if (fieldValue instanceof Float)
                 {
                     value = ((Float) fieldValue).doubleValue();
                 }
+                else if (fieldValue instanceof Integer)
+                {
+                    value = ((Integer) fieldValue).doubleValue();
+                }
+                else if (fieldValue instanceof Long)
+                {
+                    value = ((Long) fieldValue).doubleValue();
+                }
+                else if (fieldValue instanceof BigInteger)
+                {
+                    value = ((BigInteger) fieldValue).doubleValue();
+                }
+                else if (fieldValue instanceof BigDecimal)
```

Comment:
Does it make sense to warn of possible loss of precision as part of any of these conversions using messages.addMessage("Warning: ... for field: '" + fd.getFieldName() + "'");

Reply:
That is a good idea, fixed.